### PR TITLE
fix(ci): Add test coverage to CI

### PR DESCRIPTION
**Pull request by kevinmingtarja**
## Problem
 
The test coverage we had was outdated.

## Solution
 
Intergrate test coverage to our CI using coveralls, so that it is always up to date.

### CODE REVIEW

File: .github/workflows/ci-dgraph-test-coverage.yml
Comment (skrdgraph) [inline .github/workflows/ci-dgraph-test-coverage.yml:2]: I fixed this last evening, we are doing double triggers. 

you may want to change it to something like this https://github.com/dgraph-io/dgraph/pull/8241

Comment (kevinmingtarja) [reply .github/workflows/ci-dgraph-test-coverage.yml]: Got it, I've changed it in my latest commit. Thanks @skrdgraph

Comment (skrdgraph) [reply .github/workflows/ci-dgraph-test-coverage.yml]: resolving this conversation

**Comment by skrdgraph** [comment]
we should discuss about this PR https://github.com/dgraph-io/dgraph/pull/8167 - as it is kind of related to your changes

File: t/t.go
Comment (skrdgraph) [inline t/t.go:57]: this is a good choice @kevinmingtarja ...

File: t/t.go
Comment (skrdgraph) [inline t/t.go:215]: @kevinmingtarja - this will work only when we are running the tests in serial mode. This will fail when we parallelize the tests.

I think we should talk about this tomorrow, but I won't block this merge. My main reason why this should still go in, is as follows: we know for sure that our test-harness has not been stabilized for parallelization. So I will be adding a comment here in your PR, to fix this later.

File: .github/workflows/ci-dgraph-tests.yml
Comment (skrdgraph) [inline .github/workflows/ci-dgraph-tests.yml]: we should use this https://github.com/marketplace/actions/coveralls-github-action

I will try to make the change on top of your PR for this

Comment (skrdgraph) [reply .github/workflows/ci-dgraph-tests.yml]: The coveralls seems to not have cover profile option for some reason.

File: t/t.go
Comment (skrdgraph) [inline t/t.go:730]: @kevinmingtarja - we should change to this https://gobyexample.com/writing-files later on. I foresee a lot of issues.

**Comment by skrdgraph** [comment]
@kevinmingtarja - I am approving this PR, and will merge it once the tests pass. We have a little more work to do to make this cleaner. We will do that in the next PR.

**Review by skrdgraph — APPROVED** [review]

**Pull request merged**
